annotated_test.go

```diff
@@ -65,6 +67,196 @@ func TestAnnotated(t *testing.T) {
 	})
 }
 
+type asStringer struct {
+	name string
+}
+
+func (as *asStringer) String() string {
+	return as.name
+}
```

Comment:
for one of the test cases, we should also test the case where a value type implements the interface instead of the pointer type.

```
type anotherStringer struct{ name string }

return (a anotherStringer) String() { return a.name }
```